Report indicator nara

w-jan opened the pull request:
Hello Floris,

I want to publish a report about how the drought-indicator, NARA-version, is constructed. If you have a lot of time to review it, that would be very fine, but I think it's not possible.

I would also ask Thierry as reviewer, but the reviewers-list is very short. But probably the same case I think ;-)

Greetings,
Jan

w-jan committed:
snapshot package updates

w-jan committed:
start bookdown from INBOmd-template

w-jan committed:
start bookdown from inbomd-template. Organizing  files

w-jan committed:
setting report title & authors

w-jan committed:
using renv

w-jan committed:
setup

w-jan committed:
getting started

w-jan committed:
rapport klaar voor revisie

w-jan committed:
problems creating pdf:
changing {text} into {text style} -> seems to be working
another problem (creating plots) still exists

w-jan committed:
some resulting figures

w-jan committed:
add report-map to gitignore, except final outputs (html, epub and/or pdf)

w-jan committed:
resulting epub (with TEX-warnings)

w-jan committed:
new output, figures with smaller resolution

w-jan committed:
update renv snapshot

w-jan committed:
gitignore

w-jan committed:
files needed for creating a bookdown from an inbo-template

w-jan committed:
update gitignore

w-jan committed:
progression in report writing
making a reviewable version

w-jan committed:
results dd 2021.06.10

w-jan committed:
changing names files droogtemeetnet according inbo name convention

w-jan committed:
renaming files in the update-folder

florisvdh commented:
Indeed it's not easy for me to review this on short term; it seems quite an involved report. If you attach the rendered version in a message here, I may be able to at least have a quick look at that.

w-jan commented:
Hello Floris,

I understand completely. Of course, you don't have to feel obligated. The
pdf-version is attached. It's a bit a pity I can't do that for the
html-version, because there are js-scripts involved.

Greetings,
Jan

Op do 10 jun. 2021 om 20:20 schreef Floris Vanderhaeghe <
***@***.***>:

> Indeed it's not easy for me to review this on short term; it seems quite
> an involved report. If you attach the rendered version in a message here, I
> may be able to at least have a quick look at that.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/inbo/n2khab-droogtemeetnet/pull/13#issuecomment-858867357>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEIOMT47TVWU2LDUIXHAWPDTSD67XANCNFSM46O3HTSQ>
> .
>


-- 
*Jan Wouters*
Onderzoeker

Vlaamse Overheid
INSTITUUT VOOR NATUUR- EN BOSONDERZOEK
Team Milieu en Klimaat
T +32 471 33 93 94

Herman Teirlinckgebouw, Havenlaan 88 bus 73, B-1000 Brussel
www.inbo.be
////////////////////////////////////////////////////////////
///////////////////////////////

ThierryO commented on indicator-nara/00-setup.Rmd:
```diff
@@ -0,0 +1,68 @@
+#welke data verversen
+# 2: alles; 1: enkel de huidige google-drive-bestanden downloaden ; 0: enkel met lokale data
+if (params$refresh_data == 2) {
+  datapath <- "G:/Mijn Drive/PRJ_Meetnet_Droogte/2_Uitvoering/data"
```
don't use absolute paths

w-jan replied:
In this case i can't avoid it, i think : it's a reference to a path mandatory outside the repo

ThierryO commented on indicator-nara/.gitignore:
```diff
@@ -39,18 +41,23 @@ rsconnect/
 data/result/meetnet/GIS/*
-data/result/indicator_nara/GIS/*
-data/result/indicator_meetnet/GIS/*
+data/result/indicator-nara/GIS/*
```
We created some [naming conventions](https://packages.inbo.be/checklist/reference/check_filename.html) for file and folders. They require underscores instead of dashes.

w-jan replied:
done

ThierryO commented on indicator-nara/00-setup.Rmd:
```diff
@@ -0,0 +1,68 @@
+options(stringsAsFactors = FALSE)
+
+#belangrijke opmerking: run dit programma in de 64-bit versie van R
+#Tijdens de uitvoering van het programma in de 32-bit versie van R ontstonden er op mijn laptop geheugenproblemen.
```
INLA is 64-bit only

w-jan replied:
that explains everything ;-)

ThierryO commented on indicator-nara/00-setup.Rmd:
```diff
@@ -0,0 +1,68 @@
+# 2: alles; 1: enkel de huidige google-drive-bestanden downloaden ; 0: enkel met lokale data
+if (params$refresh_data == 2) {
+  datapath <- "G:/Mijn Drive/PRJ_Meetnet_Droogte/2_Uitvoering/data"
+  dir.create(file.path(datapath, "GIS/VoorR/"), recursive = TRUE)
```
```suggestion
  dir.create(file.path(datapath, "GIS", "VoorR"), recursive = TRUE)
```

w-jan replied:
done

ThierryO commented on indicator-nara/00-setup.Rmd:
```diff
@@ -0,0 +1,68 @@
+}
+
+# lokale gitignored paden aanmaken
+if (!dir.exists(file.path("..", "data", "local"))) dir.create(file.path("..", "data", "local"), recursive = TRUE)
```
You can simply this

```suggestion
dir.create(file.path("..", "data", "local"), recursive = TRUE, showWarnings = FALSE)
```

w-jan replied:
done

ThierryO commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+# Inleiding {#h1:inleiding}
+Driekwart van de voor Vlaanderen en Europa meest waardevolle natuurtypes, steunen voor hun voortbestaan op grondwater. Grondwaterafhankelijke vegetaties komen voor in en op de rand van kwelgebieden, beekvalleien, vijvers en poelen. 
+De vochtige gebieden vertonen vaak veel gradiënten en herbergen daarom een hoge biodiversiteit (De Becker, 2020). 
```
manual citation

w-jan replied:
Ref included in the bib_file

ThierryO commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+Slagen die soorten er niet in om zich aan te passen of om hun leefgebied te verschuiven, dan dreigen ze te verdwijnen. 
+Dat risico wordt groter als bijvoorbeeld het landgebruik of het grondwaterregime in de omgeving de beschikbare ruimte om uit te wijken beperkt. 
+
+# Monitoring droogte in Vlaanderen {#h2:monitoring}
```
I recommend for future documents to use only a sensible name, which can include the heading level. Heading numbers can change in a document. Then you either have to update all the labels and the references or keep a mismatch between label name and the actual heading level.

w-jan replied:
Ok good suggestion
Done

ThierryO commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+Op schaal Vlaanderen is dit een relatief laag aantal. 
+Wil men op deze schaal veralgemeenbare uitspraken kunnen maken, moet de keuze van deze punten zo goed mogelijk ruimtelijk gebalanceerd gebeuren. 
+Ook moet er rekening gehouden worden met mogelijke verschillen in het gedrag van het grond/oppervlaktewaterpeil t.a.v. een neerslagtekort. 
+@RN6212 geven een toelichting hoe dit meetnet concreet werd ontworpen. 
```
I recommend to use human readable labels for citations.

w-jan replied:
done

ThierryO commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+Drie soorten indicatoren werden hiervoor in beschouwing genomen:
+
+- een *cumulatieve*: **de som per jaar van de diepte van het grondwaterpeil beneden een kritische drempelwaarde** 
+
```
Don't add whitespace between item of the same list. Otherwise they become separate lists.

w-jan replied:
thnxs,
done

ThierryO commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+Bij het bepalen van de referentietoestand spelen twee factoren een rol:
+
+- welk grondwaterpeil beschouwen we als indicatief voor droogte oftewel wat is een **kritische waarde**?
+
```
Don't add whitespace between item of the same list. Otherwise they become separate lists.

w-jan replied:
ok, good to know !

ThierryO commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+Ze moet dus toelaten om op jaarbasis de algemene droogte-toestand van verdrogingsgevoelige vegetaties weer te geven. 
+Drie soorten indicatoren werden hiervoor in beschouwing genomen:
+
+- een *cumulatieve*: **de som per jaar van de diepte van het grondwaterpeil beneden een kritische drempelwaarde** 
```
Consider switch the bold and italics. The emphasis should be on the name not the explanation.

w-jan replied:
you're right (again;-)

ThierryO commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+Een absolute indicator laat dat wel toe.
+Zo ligt, zoals hoger al werd aangehaald, het referentieniveau of baseline bij een absolute indicator o.b.v. 5% drempelwaarde op 18 dagen. Een jaar met meer dan 18 droge dagen kan als een relatief droog jaar worden beschouwd.
+
+Conclusie:
```
Ending a line with two space forces a linebreak.
```suggestion
Conclusie:  
```

w-jan replied:
ok, thxs for this tip

ThierryO commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+Het wordt complexer als we inzicht willen krijgen in de onzekerheid die er op de droogte-indicator zit, m.a.w. wat is het betrouwbaarheidsinterval van de droogte-indicator ? 
+
+Hiervoor doen we een beroep op de formule van Rubin [@RN6220].
+We geven eerst de inhoudelijke betekenis, Vooraleer we de mathematische uitwerking geven.
```
```suggestion
We geven eerst de inhoudelijke betekenis, vooraleer we de mathematische uitwerking geven.
```

ThierryO commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+## Berekenen van de droogte-indicator {#h3-3:indicator}
+
+Kennen we de kritische drempelwaarde dan kan de droogte-indicator vrij eenvoudig berekend worden.
+Eerst tellen we voor elke meetlocatie, elk jaar en elke simulatie het aantal dagen dat het grondwaterpeil onder de kritische drempelwaarde is $=dagen\_droog_{meetlocatie_m; simulatie_s; jaar_j}$. 
```
I prefer formulas with single letter variables instead of word as variable names.
```suggestion
Eerst tellen we voor elke meetlocatie $m$, elk jaar $j$ en elke simulatie $s$ het aantal dagen $d_{mjs}$ dat het grondwaterpeil onder de kritische drempelwaarde is. 
```

w-jan replied:
thxs for the suggestion
I've abbreviated all the names in the formulas to a single letter or a few letters

ThierryO commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+$$\log(\overline{{\text{droogte-ind-sim}}}_{j;m;s}) = \beta_0 + \beta_1T_j + t_j + b_m + \epsilon_{jms}$$
+$\beta_0$ is de intercept
+
+$\beta_1T_j$ is het specifiek effect/bijdrage van een jaar
```
$\beta_1T_j$ assumes a linear trend along $T_j$.
```suggestion
$\beta_j$ is het specifiek effect/bijdrage van jaar $j$
```

w-jan replied:
thnxs for the correction !

ThierryO commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+In tegenstelling met ${\text {droogte-ind}}_{j}$ worden aan die berekening voorwaarden gekoppeld, namelijk dat het effect van een jaar afhangt van de vorig(e) jaar/jaren volgens een tweede orde functie en ook dat een meetpunt een vaste (maar onbekende) bijdrage levert (dus dat meetpunten systematisch droger of natter kunnen zijn).
+$${\text{dagen_droog}}_{j;m;s} \sim NB(\overline{{\text{droogte-ind-sim}}}_{j;m;s}, \theta)  $$
+
+$$\overline{{\text{droogte-ind-sim}}}_{j;m;s} = E({\text{dagen_droog}}_{j;m;s}) \sim {\text {jaar}_j} + (1 \vert{\text{meetlocatie}_m})  $$
```
this line is confusing

w-jan replied:
rewrote the sentence

ThierryO commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+
+Per jaar (en simulatie) wordt een gemiddelde waarde met haar standaardafwijking berekend. 
+In tegenstelling met ${\text {droogte-ind}}_{j}$ worden aan die berekening voorwaarden gekoppeld, namelijk dat het effect van een jaar afhangt van de vorig(e) jaar/jaren volgens een tweede orde functie en ook dat een meetpunt een vaste (maar onbekende) bijdrage levert (dus dat meetpunten systematisch droger of natter kunnen zijn).
+$${\text{dagen_droog}}_{j;m;s} \sim NB(\overline{{\text{droogte-ind-sim}}}_{j;m;s}, \theta)  $$
```
```suggestion
$${\text{dagen_droog}}_{j;m;s} \sim NB(\mu_{j;m;s}, \theta)  $$

$$\log(\mu_{j;m;s})} = \eta_{j;m;s})}$$
```

w-jan replied:
done

ThierryO commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+$$\overline{{\text{droogte-ind-sim}}}_{j;m;s} = E({\text{dagen_droog}}_{j;m;s}) \sim {\text {jaar}_j} + (1 \vert{\text{meetlocatie}_m})  $$
+of 
+
+$$\log(\overline{{\text{droogte-ind-sim}}}_{j;m;s}) = \beta_0 + \beta_1T_j + t_j + b_m + \epsilon_{jms}$$
```
```suggestion
$$\eta_{j;m;s}) = \beta_0 + \beta_1T_j + t_j + b_m + \epsilon_{jms}$$
```

ThierryO commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+$t_j$ is een term ('random walk') die toelaat om rekening te houden met het feit dat het effect van het ene jaar mogelijk afhankelijk is van dat van één of meer van de voorgaande jaren.
+
+Hier beschrijven we die mogelijke afhankelijkheid als een twee-orde relatie:
+$$ \Delta^2 t_i = \Delta t_{i+1} - \Delta t_i = t_{i+1} - 2t_i + t_{i-1}$$
```
```suggestion
$$ \Delta^2 t_i = \Delta t_{i+1} - \Delta t_i = t_{i+1} - 2t_i + t_{i-1} = (t_{i+1} - t_i) - (t_i - t_{i-1})$$
```

w-jan replied:
thnxs

ThierryO commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+$$ \Delta^2 t_i = \Delta t_{i+1} - \Delta t_i = t_{i+1} - 2t_i + t_{i-1}$$
+$$ \Delta^2 t_i \sim  N(0,\sigma^2)$$
+
+De term  $\beta_1T_j + t_j =  \text{Trend}_j$ beschrijft de trend .
```
```suggestion
De term  $\beta_j + t_j =  \text{Trend}_j$ beschrijft de trend .
```

ThierryO commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+Alleen meetpunten waarbij de grondwaterpeilen geen trend vertonen (cf. verdroging of vernatting) werden geselecteerd.
+Daarom kunnen we ook aannemen dat, minstens voor de referentieperiode, de locatie-effecten stabiel in de tijd zullen zijn.
+
+$\epsilon_{jms}$ is de term voor de random ruis in de log-schaal.
```
The negative binomial distribution doesn't have as noise term.

w-jan replied:
I reformulated it, but i'm not sure this is correct:
"ϵ_jms  is de term die de fout geeft tussen de verwachte en waargenomen waarde. We nemen aan dat deze fout in de log-schaal een normale verdeling volgt."

ThierryO commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+Voor het bepalen van de onzekerheid, de betrouwbaarheidsintervallen, van de trendwaarden doen we opnieuw een beroep op de formule van Rubin [@RN6220].
+Om de onzekerheid in de trend te kunnen inschatten afkomstig van het schatten van peilen, wordt er niet één model berekend, maar wordt een model voor elke simulatie berekend.
+
+$${\text{de trend van de droogte-indicator berekend voor één jaar : }} \overline{{\text {droogte-ind}}}_{j} =  \sum_{s = 1}^{20}(\overline{{\text {droogte-ind-sim}}}_{j;m;s} / 20 $$
```
```suggestion
de gemiddelde droogte-indicator voor jaar $j$:  $\overline{{\text {droogte-ind}}}_{j} =  \sum_{s = 1}^{20}(\overline{{\text {droogte-ind-sim}}}_{j;m;s} / 20 $
```

ThierryO commented on indicator-nara/022-methode-pdf.Rmd:
```diff
@@ -0,0 +1,115 @@
+## Berekenen van de droogte-indicator {#h3-3:indicator}
```
why a different file for pdf?

ThierryO commented on indicator-nara/031-berekening.Rmd:
```diff
@@ -0,0 +1,707 @@
+#hulpfunctie voor het maken van minor ticks in de grafieken
+insert_minor <- function(major_labs, n_minor) {
+  labs <- c( sapply( major_labs, function(x) c(x, rep("", 4) ) ) )
+  labs[1:(length(labs) - n_minor)]}
```
suppose that `n_minor` is larger than `length(labs)`. Then you get something like `1:-1`, which in not what you want. `seq_len(-1)` is safer as it returns `integer(0)`.

```suggestion
  labs[seq_len(length(labs) - n_minor))]}
```

ThierryO commented on indicator-nara/031-berekening.Rmd:
```diff
@@ -0,0 +1,707 @@
+    inner_join(dagen_droog_lang_breed)
+  
+  p <- ggplot(data = PI_data, aes_string(y = respons, x = "jaar"))
+  p <- p + xlab("Jaar") + ylab("Aantal dagen onder kritsche drempelwaarde")
```
I find the ggplot style like you use in the next function easier to debug.

w-jan replied:
ok

ThierryO commented on indicator-nara/031-berekening.Rmd:
```diff
@@ -0,0 +1,707 @@
+#plot van de rollende gemiddelden
+plotma.basis <- function(indic_finaal, indicator, intercept) {
+  p <- ggplot(data = indic_finaal ) + 
+    geom_line(data = indic_finaal %>% filter(jaar >= 1985 + tijdsvenster), aes(x = jaar, y = lcl), linetype = "longdash") +
```
I think it can more efficient / readable if you create a new variable `period = jaar >= 1985 + tijdvenster`. Then you can add `alpha = period` to the `aes()`.

w-jan replied:
This didn't work: Error: geom_path: If you are using dotted or dashed lines, colour, size and linetype must be constant over the line

ThierryO commented on indicator-nara/031-berekening.Rmd:
```diff
@@ -0,0 +1,707 @@
+  if (i == 1) {
+    ruwedata <- get(sprintf("ruwedata%02d",i))
+  } else {
+    ruwedata <- bind_rows(ruwedata, get(sprintf("ruwedata%02d",i)))
```
I think you can replace this entire for loop with the single line below. Untested.
```
map_df(sprintf("ruwedata%02d", 1:23), get)
```

w-jan replied:
yes, it works !

ThierryO commented on indicator-nara/031-berekening.Rmd:
````diff
@@ -0,0 +1,707 @@
+
+```{r tubes-network, include=FALSE}
+#laatste versie met de geselecteerde meetpunten ophalen. De ruwe data bevatten namelijk ook nog meetreeksen van punten van een eerdere versie.
+gitroot <- rprojroot::find_root(rprojroot::is_git_root)
````
```suggestion
gitroot <- repository()
```

w-jan replied:
ok, good to know !

ThierryO commented on indicator-nara/031-berekening.Rmd:
````diff
@@ -0,0 +1,707 @@
+```{r tubes-network, include=FALSE}
+#laatste versie met de geselecteerde meetpunten ophalen. De ruwe data bevatten namelijk ook nog meetreeksen van punten van een eerdere versie.
+gitroot <- rprojroot::find_root(rprojroot::is_git_root)
+tubes_selected <- read_vc(file.path(".","data","result","meetnet","tubes_selected"), root = gitroot)
````
`"."` is redundant
```suggestion
tubes_selected <- read_vc(file.path("data", "result", "meetnet", "tubes_selected"), root = gitroot)
```

w-jan replied:
ok

ThierryO commented on indicator-nara/031-berekening.Rmd:
```diff
@@ -0,0 +1,707 @@
+#wegschrijven in een RDS-bestand
+saveRDS(absperc_gw,file.path("cache", "percentielen.rds" ))
+
+write_vc(absperc_gw, file.path("data", "result", "percentielen-1985-2014"), sorting = c("meetpunt"), strict = FALSE)
```
Use `strict = FALSE` only when the definition of the variable change. You might want to set the root as well.

```suggestion
write_vc(absperc_gw, file.path("data", "result", "percentielen-1985-2014"), sorting = c("meetpunt"))
```

w-jan replied:
OK, i skipped the strict-argument and added a root-argument (more secure use)

ThierryO commented on indicator-nara/031-berekening.Rmd:
```diff
@@ -0,0 +1,707 @@
+Een probleem is dat één belangrijke conditie voor het gebruik van een verdeling wordt geschonden. 
+De 'trekkingen' gebeuren namelijk niet onafhankelijk van elkaar: de kans dat als het vandaag een droge dag is, het dat morgen of overmorgen ook zal zijn is veel groter dan 0,05 en vice versa. 
+Modellen die uitgaan van een binomiale of een andere verdeling gaan er van uit dat dit wel zo is. 
+We hebben onderzocht of in de dataset de indicator binomiaal verdeeld is door de 
```
incomplete sentence

w-jan replied:
ok, skipped that sentence

ThierryO commented on indicator-nara/032-berekening-html.Rmd:
```diff
@@ -0,0 +1,26 @@
+  summarise(droogte_ind_j = mean(dag_onder_p05)
+            ) %>% 
+  ungroup()
+
```
you can avoid the need for two versions. Have a look at `INBOmd::dyn_table()`. Another options is to use something like this

```
if (interactive() || opts_knit$get("rmarkdown.pandoc.to") == "html") {
# code for interactive (html) output
} else {
# code for static (pdf) output
}
```

w-jan replied:
Thanks for this suggestion (and all others !)
I felt the need because of two reasons: formulas with text and formatting tables 
I renamed the formulas so I could use this code to get all the code in one Rmd.

ThierryO commented on indicator-nara/032-berekening-pdf.Rmd:
```diff
@@ -0,0 +1,31 @@
+## De droogte-indicator en haar betrouwbaarheid {#h4-2:droogte-indicator}
+
+### Droogte-indicator {#h4-2-1:droogte-indicator}
```
Drop the section without the confidence intervals. 
```
\begin{sermon}
Thou shalt not publish uncertain estimates without their confidence interval.
\end{sermon}
```

w-jan replied:
;-)

w-jan replied:
Done

ThierryO commented on indicator-nara/035-berekening.Rmd:
````diff
@@ -0,0 +1,879 @@
+Er zit een duidelijke neerwaartse trend in.
+Deze trend is te verklaren door de toegenomen meetintensiteit.
+
+```{r droogte-indicator-imput-fig, fig.cap="Het aandeel van imputatie in de totale variatie van de droogte-indicator", message=FALSE}
````
I don't understand this figure.

w-jan replied:
I have changed the label of the y-axis: "aandeel variatie door imputatie in de totale variatie (%)". Is it more clear ?
I want to show that in the beginning much more data needed to be imputated and i want to show the cost of it. 
Also, with this figure one can estimate what the benefit is of a good datacollection-policy (no imputations). I'm a right ?

ThierryO commented on indicator-nara/035-berekening.Rmd:
```diff
@@ -0,0 +1,879 @@
+De andere verklarende factor is de locatie (de ene plaats kan een afwijkend gedrag vertonen t.o.v. een andere).
+We onderzoeken hier alleen het tijdsaspect. 
+Een mogelijke andere interessante vraag is of er regionale verschillen zijn tussen de waarden en waardoor deze dan  kunnen verklaard worden. 
+Om deze vraag te kunnen beantwoorden is het huidige en voorziene meetnet echter minstens een orde te klein!
```
```suggestion
Om deze vraag te kunnen beantwoorden is het huidige en voorziene meetnet echter minstens een grootteorde te klein!
```

florisvdh commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+## Droogtemeetnet {#h2-1:meetnet}
+Om in Vlaanderen een globaal beeld te krijgen van de evolutie van de grondwaterstanden op middellange tot lange termijn in verdrogingsgevoelige vegetaties werd een meetnet van een honderdtal punten ontworpen. 
+Op schaal Vlaanderen is dit een relatief laag aantal. 
+Wil men op deze schaal veralgemeenbare uitspraken kunnen maken, moet de keuze van deze punten zo goed mogelijk ruimtelijk gebalanceerd gebeuren. 
```
```suggestion
Wil men op deze schaal veralgemeenbare uitspraken kunnen maken, dan moet de keuze van deze punten aselect gebeuren. 
```

'Ruimtelijk gebalanceerd' zorgt voor een meer efficiënte schatter, maar niet voor een onvertekende schatter. Eventueel kun je het argument van de zin, ivm balancering, ook zo stellen (minder meetpunten nodig voor eenzelfde precisie). 'Veralgemeenbaar naar Vlaanderen' is het echter niet op statistische grond (idem voor betekenis betrouwbaarheidsintervallen); dat blijft een interpretatieslag, omdat een belangrijk deel van de steekproef een judgment sample is (bestaande meetpunten). Je kunt wel stellen dat we met het meetnet een interpretatie willen doen op Vlaams niveau.

w-jan replied:
Je hebt voor het grootste deel gelijk. Een ruimtelijke balancering gebaseerd op een vaste set punten, kan nog steeds ernstig gebiased zijn (bijv. als die set niet representatief is voor de variabele die je wenst te onderzoeken).
Ik vind wel een ruimtelijke balancering een goede techniek om representatief te kunnen zijn. Volgens mij is het een nodige voorwaarde, maar daarom nog niet een voldoende voorwaarde. In die zin zal ik de zin nog nuanceren.
Stellen dat een representatief staal enkel kan met een aselecte trekking, vind ik niet helemaal juist. Een aselecte trekking kan per toeval ook totaal 'verkeerd' (gebiased) zijn. Denk ik toch, kan me vergissen

Voor het meetnet hebben we niet uitsluitend beroep gedaan op bestaande punten. We zijn uitgegaan van de dekking aan grondwatertypen in een hok. Bij het selecteren van percelen in dat hok hebben we wel voorrang gegeven aan bestaande meetpunten, dus daar kan zeker een bias op zitten. Soms kwamen er geen meetpunten in voor en moet er voor een nieuw punt gezorgd worden. Hier heb ik wel nog een voorwaarde aan gekoppeld, nl. dat het in een perceel met natuurbeheer moet liggen (maakt administratief makkelijker en ook wel meer garantie op duurzaamheid).

ThierryO replied:
Wanneer we spreken over unbiased, dan hebben we het over de _verwachte waarde_, niet over een individuele steekproef. Geen enkele individuele steekproef zal perfect unbiased zijn. Enkel aselecte (of systematische) steekproeven garanderen een onvertekende verwachte waarde.

Als je de som van 5 dobbelstenen als variabele neemt, dan is de verwachte waarde 17.5. Iets wat je in de praktijk nooit zal tegenkomen. Als je die steekproeftrekking vaak herhaald en daarvan het gemiddelde krijgt zal je bij faire dobbelstenen wel aan de verwachte waarde van 17.5 komen.

Van zodra je afwijkt van een aselecte of systematische steekproef, heb je niet langer de garantie van een onvertekende _verwachte waarde_.

w-jan replied:
Ok, een aselecte proef zou in het geval van een meetnet betekenen, dat je elk jaar opnieuw een trekking doet. Of vergis ik me ?
Dat is natuurlijk in deze concrete situatie niet echt doenbaar ;-) Maar het is wel heel goed om te weten hoe het eigenlijk best zou gebeuren.

ThierryO replied:
Neen. Je kan perfect een aselect (of systematische) steekproef perfect meerdere jaren gebruiken. Je moet enkel zorgen dat je de steekproef kan "verversen" wanneer de steekproefpopulatie wijzigt.

Voor de meeste Natura 2000 meetnetten gebruiken we een aselecte steekproef waarbij we locaties jaarlijks of driejaarlijks opvolgen.

w-jan replied:
OK,  ik was verkeerd door te zeggen dat door een aselecte trekking je uitkomst kan gebiased zijn.  Per definitie is die steeds ungebiased. Je resultaat kan er wel nog serieus naast zitten, de kans wordt hierop wel kleiner naarmate je steekproef groter wordt. Het is de variabiliteit van die variabele die dan richtinggevend is om te bepalen hoe groot je trekking moet zijn. 
Het ruimtelijk balanceren is een hulpmiddel om die variantie wat te verkleinen (of zoals Floris schreef: de efficiëntie vergroot), net zoals door het stratifiëren op de variabelen die ook een mogelijke impact hierop hebben.
Klopt die zienswijze ?

florisvdh replied:
Het is zoals Thierry aangeeft: 1 steekproefuitkomst (en in de praktijk heb je er maar één) wijkt vaak af van de verwachte waarde van de schatter (die voor een onvertekende schatter gelijk is aan de populatieparameter). Belangrijk is dat je wel het betrouwbaarheidsinterval van die ene uitkomst nog steeds kunt interpreteren in relatie tot de verwachte waarde (met bv. 95% betrouwbaarheid). Een aselecte steekproefstrategie heb je nodig om te kunnen stellen dat de verwachte waarde van de schatter gelijk is aan de populatieparameter, waarin we zijn geïnteresseerd.

De keuze van welke aselecte steekproefstrategie je gebruikt heeft invloed op de distributie van de schatter (distributie volgens herhaalde fictieve trekkingen), en dus op de breedte van bovenvermeld betrouwbaarheidsinterval (in de veronderstelling dat je dit voordeel benut in de berekening van varianties). Steekproefstrategie heeft dus sowieso invloed op de afwijking van de steekproefuitkomst tov de echte waarde. Een schatter volgens GRTS is efficiënter dan volgens simple random sampling: dwz zijn distributie is smaller.

> een aselecte proef zou in het geval van een meetnet betekenen, dat je elk jaar opnieuw een trekking doet

In principe is de spatiale doelpopulatie (verspreiding van de locaties van interesse) continu in verandering, dus idealiter heb je continue updates van het volledige steekproefkader. Dan zou je in theorie jaarlijks (maar in praktijk bv. zesjaarlijks) opnieuw een trekking doen om de steekproefstrategie te blijven respecteren. Een GRTS-master heeft het gigantische voordeel dat het merendeel van de steekproefelementen tussen die trekkingen gemeenschappelijk zullen zijn. Dus ja, opnieuw trekken is in principe de bedoeling, in de praktijk komt dit neer op beperkte wijzigingen van de steekproef. Dit is trouwens in overeenstemming met het antwoord van Thierry. Uiteindelijk hangt veel af van de snelheid waarmee de doelpopulatie wijzigt, om te kunnen oordelen welke termijn nog redelijk is om een steekproef vast te houden. Een perfect steekproefkader hebben we overigens quasi nooit, dus daar komt al een ongekende vertekening insluipen. Het zo goed mogelijk krijgen of houden van het steekproefkader is dan ook geen overbodige luxe om een zinvolle steekproef te kunnen nemen.

NB. het woord 'representatief' heeft allerlei betekenissen en wordt beter vermeden voor statistische context. Zie bv.:

- Kruskal W. & Mosteller F. (1979). Representative Sampling, III: The Current Statistical Literature. International Statistical Review / Revue Internationale de Statistique 47 (3): 245–265. https://doi.org/10.2307/1402647.
  - _Citation on p. 259: "Representative sampling is a term easier to avoid because it is disappearing from the technical vocabulary. At different times it has been used for random sampling, proportionate sampling, quota sampling, and purposive sampling. In general, if often denotes the aims of representing a population well with a sample"_
- Kruskal W. & Mosteller F. (1980). Representative Sampling, IV: The History of the Concept in Statistics, 1895-1939. International Statistical Review / Revue Internationale de Statistique 48 (2): 169–195. https://doi.org/10.2307/1403151.
- p. 186 in: Quataert P. (2011). Methodological and statistical aspects of indices of biotic integrity to assess the ecological condition of surface waters. PhD theses of the Research Institute for Nature and Forest, Nr. INBO.T.2011.1. Instituut voor Natuur- en Bosonderzoek (INBO), Brussel, 218 p.
- "Over de betekenissen van ‘representatief’ in de KRW-literatuur": https://drive.google.com/file/d/1KF3MNZ0P93hkMcEakzl-eXxEop_6i4Ta/view?usp=sharing

w-jan replied:
Floris, bedankt voor de toelichting! Ik denk dat ik het begrijp.
De tekst werd herschreven: veralgemeenbaarheid en representatief worden geschrapt en vervangen door efficiëntie

florisvdh replied:
OK. 'We pogen zo goed mogelijk te weerspiegelen' is ook een van de vele mogelijke manieren om die gedachte te beschrijven. Maar belangrijker is dat de kenmerken van de methode beschreven zijn - dat zijn ze via het achterliggend rapport.

florisvdh commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+
+- een *relatieve*: **aantal relatief droge dagen per jaar** 
+Een relatief droge dag is een dag waarbij het grondwaterpeil lager staat dan een bepaalde percentielwaarde voor die dag in een bepaalde periode.
+We gaan verder in dit hoofdstuk wat dieper in op de beweegredenen om  voor de beleidsrapportering een absolute indicator te verkiezen. 
```
Moet deze zin niet op een nieuwe alinea beginnen? (whiteline) Hangt nu bij de relatieve indicator.

w-jan replied:
ok, klopt idd

florisvdh commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+Dit bemoeilijkt het werken met tijdreeksen van dagelijkse metingen. 
+
+Het tijdsvenster is een andere belangrijke factor bij het vastleggen van een referentie.
+Men heeft dan te bepalen of gewerkt wordt met een vast tijdskader (van jaar a tot jaar b) of een relatief tijdkader (bijv. de laatste x jaren).
```
```suggestion
Men heeft dan te bepalen of gewerkt wordt met een vast tijdskader (van jaar a tot jaar b) of een relatief tijdskader (bijv. de laatste x jaren).
```

florisvdh commented on indicator-nara/01-inleiding.Rmd:
```diff
@@ -0,0 +1,109 @@
+Een periode van dertig jaar is gebruikelijk in klimaatstudies. 
+De reden dat we voor een vast venster kiezen, is dat de trend gemakkelijker te interpreteren is. 
+Ook het KMI werkt met vaste vasters, die ze periodiek herziet.
+Het venster kan na bijv. vijftien jaar worden herzien: dus bijv. vanaf 2035 kan 2000 - 2029 als nieuw referentie genomen worden).
```
```suggestion
Het venster kan na bijv. vijftien jaar worden herzien: dus bijv. vanaf 2035 kan 2000 - 2029 als nieuwe referentie genomen worden).
```

florisvdh commented on indicator-nara/022-methode-html.Rmd:
```diff
@@ -0,0 +1,116 @@
+
+Kennen we de kritische drempelwaarde dan kan de droogte-indicator vrij eenvoudig berekend worden.
+Eerst tellen we voor elke meetlocatie, elk jaar en elke simulatie het aantal dagen dat het grondwaterpeil onder de kritische drempelwaarde is $=dagen\_droog_{meetlocatie_m; simulatie_s; jaar_j}$. 
+Voor de periode 1985 - 2018, 58 meetlokaties en 20 simulaties betekent dit dus 34 * 58 * 20 = 39440 ${\text {dagen\_droog}}$-waarden
```
```suggestion
Voor de periode 1985 - 2018, 58 meetlocaties en 20 simulaties betekent dit dus 34 * 58 * 20 = 39440 ${\text {dagen\_droog}}$-waarden
```

florisvdh commented on indicator-nara/031-berekening.Rmd:
```diff
@@ -0,0 +1,707 @@
+Bij een dergelijke indicator wordt het aantal dagen per jaar dat het grondwaterpeil gelijk aan of onder een kritische drempelwaarde zakt opgeteld.
+Deze kritische drempelwaarde is specifiek voor elk meetpunt.
+Als kritische drempelwaarde wordt gekozen voor 
+**het _jaargemiddelde_ van de 5%-percentiel van de grondwatermetingen _van een jaar_ voor het meetpunt voor de periode 1985 - 2014**, cfr. \@ref(h3-2:drempelwaarde), die periodiek, bijv. elke vijftien jaar, met meer recente data kan herijkt worden, cfr. \@ref(h2-2-1:referentietoestand).
```
```suggestion
**het _gemiddelde_ van de _jaarlijkse_ 5%-percentielen van de grondwatermetingen voor het meetpunt voor de periode 1985 - 2014**, cfr. \@ref(h3-2:drempelwaarde), die periodiek, bijv. elke vijftien jaar, met meer recente data kan herijkt worden, cfr. \@ref(h2-2-1:referentietoestand).
```

w-jan replied:
idd, duidelijker. Bedankt !

florisvdh commented on indicator-nara/035-berekening.Rmd:
```diff
@@ -0,0 +1,879 @@
+Bij de rollende gemiddelden worden de pieken afgevlakt en dan is goed te zien dat naarmate door het vergroten van het tijdsvenster deze uitvlakken het tijdreeksmodel de resulterende golfbeweging steeds beter kan bijhouden.
+Het valt bij de grotere tijdsvensters ook op te merken dat de gemiddelde waarde duidelijk hoger is dan de referentie van 18 dagen. 
+Dit is volledig toe te schrijven aan de combinatie van de berekeningswijze van de kritische drempelwaarde, nl. een gemiddelde berekenen van afzonderlijke jaargemiddelden i.p.v. een gemiddelde voor het volledige tijdsvenster (cfr. \@ref(h3-2:drempelwaarde) en het feit dat range van mogelijke afwijkingen voor droge jaren veel groter is dan de range bij natte jaren.
+Daaruit kan men concluderend dat de referentie van 18 dagen enkel op jaarbasis mag gebruikt worden om te bepalen of het tijdens dat jaar al dan niet droog was.
```
```suggestion
Daaruit kan men concluderen dat de referentie van 18 dagen enkel op jaarbasis mag gebruikt worden om te bepalen of het tijdens dat jaar al dan niet droog was.
```

florisvdh reviewed (commented):
Impressive work @w-jan, and you also succeed in presenting it nicely! I have not reviewed the code but only read through the PDF. I added some comments therein: [nara-indicator-droogte_opmFV.pdf](https://github.com/inbo/n2khab-droogtemeetnet/files/6726340/nara-indicator-droogte_opmFV.pdf)

Minor (mainly textual) suggestions are below, as well as:

- `bib-jan-wouters.bib`: a large file (in terms of code); if edits are necessary, they cannot be done in this PR since GitHub finds it too large to show the file (note: see inbo/styles#1). Usually it's better to restrict a bib file to the references used.
- I didn't look at the specific differences, but several Rmds seem to have > 1 copy (html vs pdf), which is difficult to manage (making updates etc). It could be more useful to control behaviour with e.g. `knitr::is_html_output()` where needed, and maintain 1 copy instead.

w-jan committed:
renaming files according Inbo convention

w-jan committed:
update document after review

w-jan committed:
new bib-file with only the cited sources

w-jan committed:
update text with comments from Katrijn Alaerts

w-jan committed:
uncomplete update after review Thierry and Floris (not completed)

w-jan committed:
update regarding to the review Floris, Katrijn & Thierry

w-jan committed:
update resulting data & figures

w-jan committed:
ignore report-folder

w-jan committed:
added calculation of a 'wet-indicator': monitoring the count of days the groundwater level exceeds a high level (= mean p95-percentile over the period 1985 - 2014)

w-jan committed:
resulting data of the new calculation

florisvdh commented on indicator_nara/01_inleiding.Rmd:
```diff
@@ -1,71 +1,94 @@
+
+Men mag het tijdelijk karakter echter niet onderschatten.
+Om de droogte in beeld te brengen volstaat immers niet alleen een beeld te geven van het neerslagtekort. 
+Hoewel het neerslagtekort een drijvende kracht is voor het optreden van grondwaterdalingen, zijn er andere factoren die meespelen hoe dat neerslagtekort zich effectief in peilverlagingen zal vertalen. 
```
Hier introduceer je voor het eerst je focus op grondwater, maar doet dit m.i. veeleer impliciet. Ik zou hoger een zin toevoegen dat het doel van deze monitoring en dus van de indicator is om (acuut) zicht te krijgen op grondwaterstandsdalingen, gezien het belang hiervan voor grondwaterafhankelijke natuur. Momenteel staat dit nog niet vermeld. Het is volgens mij het argument waarom je niet alle facetten van verdroging, alsook niet-grondwaterafhankelijke natuur, onderzoekt.

w-jan replied:
Ik heb de inleiding in die zin herschikt.

florisvdh replied:
Prima :smile_cat:  - hoewel ik kan het niet zien (staat nog lokaal wellicht) :yum:

florisvdh commented on indicator_nara/01_inleiding.Rmd:
```diff
@@ -1,71 +1,94 @@
+Hoewel het neerslagtekort een drijvende kracht is voor het optreden van grondwaterdalingen, zijn er andere factoren die meespelen hoe dat neerslagtekort zich effectief in peilverlagingen zal vertalen. 
+De periode van het jaar is een eenvoudig voorbeeld van een dergelijke factor: een periode zonder neerslag zal in de winter een ander effect hebben dan in de zomer. 
+Zo kan niet gesteld worden dat een korte duur van een neerslagtekort, altijd slechts maar tot kortstondige peilverlagingen zal leiden. 
+Een periode van neerslagslagtekort kan zich over een langere tijd in de lagere (grond)waterpeilen uitsmeren.
```
```suggestion
Een periode van neerslagslagtekort kan over een langere tijd lagere (grond)waterpeilen teweeg brengen.
```

Wel even checken of dit overeenkomt met wat je bedoelde.

florisvdh commented on indicator_nara/01_inleiding.Rmd:
```diff
@@ -1,71 +1,94 @@
+@Jansen2020 geven een overzicht van de impact van de recente droge jaren op de natuur in Nederland en geven hiervoor ook causale verbanden. 
+
+
+We spreken van verdroging als door structurele oorzaken er op langere termijn een systematische daling van het grondwaterpeil wordt waargenomen of ook wanneer dit grondwater van samenstelling verandert.
```
```suggestion
We spreken in de context van het droogtemeetnet en dus dit rapport van verdroging als door structurele oorzaken er op langere termijn een systematische daling van het grondwaterpeil wordt waargenomen of ook wanneer dit grondwater van samenstelling verandert als gevolg van grondwaterdaling.
```

florisvdh commented on indicator_nara/035_berekening.Rmd:
```diff
@@ -7,7 +7,7 @@ Een 'normaal' jaar is eerder uitzondering dan regel.
 Natte jaren zijn minder opvallend, omdat de droogte-indicator niet onder nul kan gaan.
 Uitgezonderd voor het jaar 2018 zit er vrij weinig variatie in de hoogte van de pieken. 
-Men ziet wel een wijziging in de frequentie ervan, met een toename vanaf rond 2000.
+We zien wel een wijziging in de frequentie ervan, met een toename vanaf rond 2000.
```
```suggestion
We zien wel een wijziging in de frequentie ervan, met een toename vanaf ca. 2000.
```

w-jan committed:
introduction: changing structure (advice Floris)
calculation: adding a method for global indicating dessication

w-jan committed:
Update indicator_nara/01_inleiding.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
Update indicator_nara/01_inleiding.Rmd

Co-authored-by: Floris Vanderhaeghe <floris.vanderhaeghe@inbo.be>

w-jan committed:
choosing mixed-model to calculatie credible intervals in stead of Rubin-formula

w-jan committed:
resolve merge-conflict

Merge branch 'report_indicator_nara' of https://github.com/inbo/n2khab-droogtemeetnet into report_indicator_nara

# Conflicts:
#	indicator_nara/01_inleiding.Rmd

w-jan committed:
new results
